### PR TITLE
Fix problem with opening YML files where the root element is an array.

### LINKED ISSUE #274: Cannot open a YML file that has a root element of an array

### PR DESCRIPTION
**Pull request by ericroberts**
Closes #274

### CODE REVIEW

File: app/controllers/base.rb [hunk 1]
```diff
@@ -88,6 +88,11 @@ def blank_is_nil!(hash)
 					end
 				end
 			end
+
+			def render_error(status_code, message)
```
Comment (singpolyma) [inline app/controllers/base.rb]: do we want to use this, or use exceptions we catch and render a standard error template?

**Comment by ericroberts** [comment]
@singpolyma not sure if rendering a 500 error for the incompatible frontmatter types is appropriate... I couldn't think of a better one. Let me know your thoughts.

File: app/controllers/file.rb [hunk 2]
```diff
@@ -275,6 +281,12 @@ def breadcrumbs
 				files = OpenStruct.new(fn: "files", route: site.route("files"))
 				[site, files] + breadcrumbs
 			end
+
+			def render_or_error
+				yield
+			rescue Tint::File::IncompatibleFrontmatter => e
```
Comment (singpolyma) [inline app/controllers/file.rb]: could use a sinatra exception handler up the stack for this?

Comment (ericroberts) [inline app/controllers/file.rb]: not sure what you mean by this?

**Comment by singpolyma** [comment]
Hmm. It's not really the user/request's fault. not a bad request of any kind. It *might* be a 409, buit really it is programmer error so 500 is likely appropriate

File: app/controllers/file.rb (hunk 2 above)
Comment (singpolyma) [inline app/controllers/file.rb]: sinatra will let you install an exception handler for requests. Instead of wrapping every handler in a rescue block

Comment (ericroberts) [inline app/controllers/file.rb]: OK I will look into it and update

**Comment by ericroberts** [comment]
@singpolyma comments addressed.

**Pull request merged**